### PR TITLE
rust: updated rust-toolchain to latest stable 1.65.0 version

### PR DESCRIPTION
**Pull request by zherendong**
- ran 'cargo clippy' and fixed warnings

### CODE REVIEW

**Comment by FawazTirmizi** [comment]
Is there a reason for this bump? And, have we decided on a policy for bumping projects using the stable toolchain? For Hubris we avoid it unless we need to, but that's because we're stuck on nightly there. I'm inclined to follow the same policy here, but I think we face less risk on stable than nightly.

**Comment by beezow** [comment]
> Is there a reason for this bump? And, have we decided on a policy for bumping projects using the stable toolchain? For Hubris we avoid it unless we need to, but that's because we're stuck on nightly there. I'm inclined to follow the same policy here, but I think we face less risk on stable than nightly.

The version bump is required to support a new version of probe-rs.  It needs at least 1.60, while we were on 1.59.  I figured if we are bumping versions, might as well go all the way to the latest stable.  I don't see a reason to stay on 1.60 over 1.65.

**Comment by zherendong** [comment]
>  Is there a reason for this bump?

Updated commit message to reflect the motivation: latest probe-rs has a dependency of 'snapbox-macros v0.3.1' which requires rustc 1.60.0 or newer.

File: humility-arch-cortex/src/scs.rs [hunk 1]
```diff
@@ -71,7 +71,7 @@ register_offs!(SWTF_CTRL, 0x0,
     pub es0, set_es0: 0;
 );
 
-#[derive(Copy, Clone, Debug, PartialEq)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
```
Comment (beezow) [inline humility-arch-cortex/src/scs.rs:74]: What lint recommended adding this extra trait?

**Comment by FawazTirmizi** [comment]
> > Is there a reason for this bump? And, have we decided on a policy for bumping projects using the stable toolchain? For Hubris we avoid it unless we need to, but that's because we're stuck on nightly there. I'm inclined to follow the same policy here, but I think we face less risk on stable than nightly.
> 
> The version bump is required to support a new version of probe-rs. It needs at least 1.60, while we were on 1.59.

That's a pretty good reason.

> I figured if we are bumping versions, might as well go all the way to the latest stable. I don't see a reason to stay on 1.60 over 1.65.

I think that's fine, especially because we're on stable.

**Review by beezow — CHANGES_REQUESTED** [review]
Looks good, we should probably change the MSRV in the CI.  Should be in `.github/workflows/ci.yaml`.
Surprised some of these things were not lints before, they will likely cause merge conflicts in the future... *sigh*

File: humility-arch-cortex/src/scs.rs (hunk 1 above)
Comment (zherendong) [reply humility-arch-cortex/src/scs.rs:74]: [derive_partial_eq_without_eq](https://rust-lang.github.io/rust-clippy/master/index.html#derive_partial_eq_without_eq)

Comment (beezow) [reply humility-arch-cortex/src/scs.rs:74]: Huh, that's a strange lint.  Basically says to derive the trait because you can.

**Review by beezow — APPROVED** [review]

**Comment by beezow** [comment]
@Edwardong Can you rebase your changes rather than merge commit?

**Comment by zherendong** [comment]
> @Edwardong Can you rebase your changes rather than merge commit?

@beezow  My bad, this was done by mistake. GitHub mobile app only provides one button after a PR is approved. I didn't mean to do so.

**Pull request merged**